### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by jacky309**

### CODE REVIEW

File: codegen/facelift/templates/IPCAdapter.template.cpp [hunk 1]
```diff
@@ -44,14 +44,28 @@
...
+
+#ifdef DBUS_IPC_ENABLED
+    {{interfaceName}}IPCDBusAdapter m_ipcDBusServiceAdapter;
+#endif
+
+};
+
+
 {{interfaceName}}IPCAdapter::{{interfaceName}}IPCAdapter(QObject* parent) : BaseType(parent) {
+    m_impl = new Impl();
```
Comment (bitmouse) [inline codegen/facelift/templates/IPCAdapter.template.cpp]: Move to constructor initializer list

File: codegen/facelift/templates/IPCAdapter.template.h [hunk 2]
```diff
@@ -74,10 +68,11 @@ class {{classExport}} {{interfaceName}}IPCAdapter: public ::facelift::IPCService
...
 
     {{interfaceName}}IPCAdapter(QObject* parent = nullptr);
+    ~{{interfaceName}}IPCAdapter();
 
-#ifdef DBUS_IPC_ENABLED
-    {{interfaceName}}IPCDBusAdapter m_ipcDBusServiceAdapter;
-#endif
+    struct Impl;
+
+    Impl* m_impl;
```
Comment (bitmouse) [inline codegen/facelift/templates/IPCAdapter.template.h]: can std::unique_ptr be used here instead?

File: codegen/facelift/templates/IPCProxy.template.cpp [hunk 3]
```diff
@@ -35,27 +35,56 @@
...
+    + 1
 #endif
-    } }
+    > m_proxies = {};
+};
+
+{{className}}::{{className}}(QObject *parent) : BaseType(parent)
+
 {
+    m_impl = std::make_unique<Impl>();
```
Comment (bitmouse) [inline codegen/facelift/templates/IPCProxy.template.cpp]: move to constructor initializer list

File: codegen/facelift/templates/IPCAdapter.template.h (hunk 2 above)
Comment (jacky309) [reply codegen/facelift/templates/IPCAdapter.template.h]: no, for some obscure reason, although it works on the IPCProxy side...

**Pull request merged**